auth/authenticators/authenticator.ts

```diff
@@ -31,7 +31,8 @@ export class Authenticator implements AuthenticatorInterface {
     }
   }
 
-  public authenticate(options: AuthenticateOptions, callback: AuthenticateCallback): void {
-    throw new Error('Should be implemented by subclass!');
+  public authenticate(options: AuthenticateOptions): Promise<void | Error> {
```

Comment:
```ts
public authenticate(options: AuthenticateOptions): Promise<void | Error>
```
or
```ts
public authenticate(options: AuthenticateOptions): Promise<void>
```

Reply:
This needs to hold the superset of the different types for the authenticators